pos_voucher/report/pos_voucher_report.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+<template id="pos_voucher_report">
+    <t t-call="web.html_container">
+        <t t-foreach="docs" t-as="o">
+            <div class="page">
+                <div class="row">
+                    <t t-set="company" t-value="res_company"/>
+                    <div style="text-align:center">
+                        <h2 t-esc="o.code"/>
+                        <span t-esc="company.name"/><br/>
+                        Tel:<span t-esc="company.phone"/><br/>
+                        <span t-esc="company.email"/><br/>
+                        <span t-esc="company.website"/><br/><br/>
+                        <span>Please Keep Your Voucher Code Safe</span><br/>
+                        <img
+                            t-att-src="'%s/report/barcode/?type=%s&amp;value=%s&amp;width=%s&amp;height=%s&amp;humanreadable=0' % (o.env['ir.config_parameter'].sudo().get_param('web.base.url'), 'Code128', o.code, 600, 100)" alt="Barcode"
```

Comment:
Why do we need to get the `web.base.url` here?
I think it even makes it incompatible with some deployments, where wkhtmltopdf relies on `report.url` instead.

Can you just do this?

```xml
<img t-att-src="'/report/barcode/?type=%s&amp;value=%s&amp;width=%s&amp;height=%s&amp;humanreadable=0' % ('Code128', o.code, 600, 100)"/>
```

Reply:
I'm having this empty barcode with this voucher:

![image](https://user-images.githubusercontent.com/1914185/69811101-1dd94400-11cc-11ea-83d5-6a5ea4506682.png)


![image](https://user-images.githubusercontent.com/1914185/69811085-131eaf00-11cc-11ea-91b0-e3a9a88fad3a.png)

I think maybe the name is not always convertable to a barcode?

What if the barcode doesn't depend on the name?
What if the barcode is generated based on the database ID. Like: "VCHR000000001" or something like that?

Reply:
okay, previously it was like this, but in my local somehow it wasn't working, so I added web.base.url.
I will revert it

Reply:
It depends on your local config. I don't remember now the reason, but sometimes the wkhtmltopdf process can't access odoo's resources, and you get that issue. But it's usually solved by setting the `report.url` system parameter to `127.0.0.1:8069` (or whatever port you're using).

Sometimes it's just the opposite, you have to remove that parameter so that Odoo uses `web.base.url` instead

Depends on your reverse proxy configuration, I think.

http://www.odooninja.com/parameter-report-url-loading-css-reports/

Anyways, thanks for reverting it :)